### PR TITLE
Hold: Discard WebP image if it is larger than corresponding JPEG image

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
## Summary

Below is the scenario when we have a WebP image more extensive than the original image.

## Scenario 1

**If WebP main and thumbnail images are `>=` the original image size, do not update sources for the image so it will not check and generate WebP for its thumbnails.**

### **Before Patch:**

WebP image bigger than original image - BEFORE PATCH

**Frontend Image HTML:**

``

### **After Patch:**

WebP image bigger than original image - AFTER PATCH

**Frontend Image HTML:**

``

## Scenario 2

**If WebP for the main(full) image is smaller than the original image and its thumbnail WebP is more extensive than the original image then the script generates only the full image WebP**

### **Before Patch:**

WebP full image small and thumb image bigger than original image - BEFORE PATCH

**Frontend Image HTML:**

``

### **After Patch:**

WebP full image small and thumb image bigger than original image - AFTER PATCH

**Frontend Image HTML:**
``

# In this test, I use the attached images given by the user in this [PR #7](https://github.com/WordPress/performance/issues/7)

image
bigimage

## Checklist

- [ ] PR has either `[Focus]` or `Infrastructure` label.
- [ ] PR has a `[Type]` label.
- [ ] PR has a milestone or the `no milestone` label.

### CODE REVIEW

File: modules/images/webp-uploads/load.php [hunk 1]
```diff
@@ -154,6 +162,15 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 				continue;
 			}
 
+			/* Remove WebP image if it bigger then original thumbnail image */
+			$original_thumbnail_image_filesize = $properties['filesize'];
+			$webp_thumbnail_image_filesize = $source['filesize'];
+			if ( $webp_thumbnail_image_filesize >= $original_thumbnail_image_filesize ) {	
+				$destination = trailingslashit( $original_directory ) . "{$source['file']}";
+				wp_delete_file_from_directory( $destination, $original_directory );
+				return $metadata;
```
Comment (felixarntz) [inline modules/images/webp-uploads/load.php:171]: This shouldn't `return` here, it should only discard _this_ current image size if needed (i.e. `continue`). If for one image size the WebP is larger and for the other ones it's smaller, then those other WebP files should still be generated.

File: modules/images/webp-uploads/load.php [hunk 2]
```diff
@@ -96,6 +96,14 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 			continue;
 		}
 
+		/* Remove WebP image if it bigger then original image */
+		$original_image_filesize = $metadata['filesize'];
+		$webp_image_filesize = $image['filesize'];
+		if ( $webp_image_filesize >= $original_image_filesize ) {
```
Comment (felixarntz) [inline modules/images/webp-uploads/load.php:102]: We should also ensure neither of these values is `0` or not set, which can be possible. In that case we can't make the comparison reliably, so we shouldn't do it.

File: modules/images/webp-uploads/load.php [hunk 3]
```diff
@@ -154,6 +162,15 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 				continue;
 			}
 
+			/* Remove WebP image if it bigger then original thumbnail image */
+			$original_thumbnail_image_filesize = $properties['filesize'];
+			$webp_thumbnail_image_filesize = $source['filesize'];
+			if ( $webp_thumbnail_image_filesize >= $original_thumbnail_image_filesize ) {
```
Comment (felixarntz) [inline modules/images/webp-uploads/load.php:168]: See above, we should only discard any images if we know the `filesize` data  for both - but not if it is `0` or not set.

File: modules/images/webp-uploads/load.php (hunk 1 above)
Comment (mukeshpanchal27) [reply modules/images/webp-uploads/load.php]: Thanks for the feedback. I removed `return` and used `continue`.

File: modules/images/webp-uploads/load.php (hunk 2 above)
Comment (mukeshpanchal27) [reply modules/images/webp-uploads/load.php:106]: Filesize value checked and `continue` if the value is `0` or not set.

File: modules/images/webp-uploads/load.php (hunk 3 above)
Comment (mukeshpanchal27) [reply modules/images/webp-uploads/load.php:176]: See above

**Comment by mukeshpanchal27** [comment]
Closing this PR as the original PR marge in the performance repo.

**Pull request closed without merging**